electron-core/src/electron/CapacitorElectronApp.ts

```diff
@@ -221,13 +221,13 @@ export class CapacitorElectronApp {
       },
     };
 
-    this.mainWindowReference = new BrowserWindow(
-      Object.assign(
-        this.config.mainWindow.windowOptions,
-        neededBrowserWindowConfig
-      )
+    const browserWindowConfig: BrowserWindowConstructorOptions = Object.assign(
+      neededBrowserWindowConfig,
+      this.config.mainWindow.windowOptions
     );
 
+    this.mainWindowReference = new BrowserWindow(browserWindowConfig);
```

Comment:
While splitting out the assign from the construction is fine, the order of the parameters provided to assign matters. The `neededBrowserWindowConfig` should be passed as the second parameter as there are settings that need to be set in order to make other features function properly such as: plugins, auto inject custom plugins, splashscreen, ect.